sled-agent/src/rack_setup/service.rs

```diff
@@ -1192,122 +1208,93 @@ impl ServiceInner {
             config.az_subnet(),
         )?;
 
-        let marker_paths: Vec<Utf8PathBuf> = storage_manager
+        let started_marker_paths: Vec<Utf8PathBuf> = storage_manager
+            .get_latest_disks()
+            .await
+            .all_m2_mountpoints(CONFIG_DATASET)
+            .into_iter()
+            .map(|p| p.join(RSS_STARTED_FILENAME))
+            .collect();
+
+        let completed_marker_paths: Vec<Utf8PathBuf> = storage_manager
             .get_latest_disks()
             .await
             .all_m2_mountpoints(CONFIG_DATASET)
             .into_iter()
             .map(|p| p.join(RSS_COMPLETED_FILENAME))
             .collect();
 
-        let ledger =
-            Ledger::<RssCompleteMarker>::new(&self.log, marker_paths.clone())
-                .await;
+        let started_ledger = Ledger::<RssStartedMarker>::new(
+            &self.log,
+            started_marker_paths.clone(),
+        )
+        .await;
+        let completed_ledger = Ledger::<RssCompleteMarker>::new(
+            &self.log,
+            completed_marker_paths.clone(),
+        )
+        .await;
 
         // Check if a previous RSS plan has completed successfully.
         //
         // If it has, the system should be up-and-running.
```

Comment:
Maybe expand or reword this comment slightly to cover the new `else` block?

Reply:
Good call. Done in [ddb7cc3](https://github.com/oxidecomputer/omicron/pull/7185/commits/ddb7cc324ca822f24e35587051cc4e78b95c7b47)